lib/Elevate/Leapp.pm

```diff
@@ -221,6 +223,41 @@ sub extract_error_block_from_output ( $self, $text_ar ) {
     return $error_block;
 }
 
+sub check_upgrade_log_for_failures ($self) {
+    my $upgrade_log = LEAPP_UPGRADE_LOG;
+    my $cont_file   = LEAPP_FAIL_CONT_FILE;
+
+    if ( !-e $upgrade_log ) {
+        ERROR("No LEAPP upgrade log detected at [$upgrade_log]");
+
+        return 1;
+    }
+
+    my $grep_firstboot = $self->cpev->ssystem_capture_output( '/usr/bin/grep', '"Starting stage After of phase FirstBoot"', $upgrade_log );
```

Comment:
FIX:  We should avoid the call to an external system binary here.  Instead, we should use `Cpanel::LoadFile` to read the file, and parse the results.

```
my $contents = Cpanel::LoadFile::loadfile($upgrade_log) // '';
my @lines = split( "\n", $contents );

return 0 if grep { $_ =~ m/Starting stage After of phase FirstBoot/ } @lines;
```

Reply:
Actually File::Slurper would be preferred as it is compatible with Test::MockFile